pcap/pcap.go

```diff
@@ -455,6 +456,7 @@ func (p *Handle) compileBPFFilter(expr string) (_Ctype_struct_bpf_program, error
 // CompileBPFFilter compiles and returns a BPF filter for the pcap handle.
 func (p *Handle) CompileBPFFilter(expr string) ([]BPFInstruction, error) {
 	bpf, err := p.compileBPFFilter(expr)
+	defer C.pcap_freecode(&bpf)
```

Comment:
You are right. What is the better way to resolve this issue:

1. Remove the `C.pcap_freecode` in function compileBPFFilter on line 449 and delegate the freeing always to the calling function (remove line 449)
2. Move the `defer`statements in CompileBPFFilter and SetBPFFilter below the error checking code